airflow/providers/amazon/aws/operators/eks.py

```diff
@@ -281,29 +282,31 @@ class EksCreateNodegroupOperator(BaseOperator):
     def __init__(
         self,
         cluster_name: str,
-        nodegroup_subnets: List[str],
+        nodegroup_subnets: Union[List[str], str],
         nodegroup_role_arn: str,
-        nodegroup_name: Optional[str] = DEFAULT_NODEGROUP_NAME,
+        nodegroup_name: str = DEFAULT_NODEGROUP_NAME,
         create_nodegroup_kwargs: Optional[Dict] = None,
         aws_conn_id: str = DEFAULT_CONN_ID,
         region: Optional[str] = None,
         **kwargs,
     ) -> None:
         self.cluster_name = cluster_name
-        self.nodegroup_subnets = nodegroup_subnets
         self.nodegroup_role_arn = nodegroup_role_arn
         self.nodegroup_name = nodegroup_name
         self.create_nodegroup_kwargs = create_nodegroup_kwargs or {}
         self.aws_conn_id = aws_conn_id
         self.region = region
+        if isinstance(nodegroup_subnets, str):
+            self.nodegroup_subnets = cast(List, literal_eval(nodegroup_subnets))
+        else:
+            self.nodegroup_subnets = nodegroup_subnets
```

Comment:
This seems like more than a mypy fix but adding a feature -- are we currently using "string-containing-a-list" feature?

Reply:
If we keep it we should do more than a cast -- we should validate the type

Reply:
Oh yes we are `        nodegroup_subnets="{{ dag_run.conf['nodegroup_subnets'] }}",`